app/controllers/backoffice/approval_requests_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class Backoffice::ApprovalRequestsController < Backoffice::ApplicationController
+  before_action :find_and_authorize, only: %i[show update]
+
+  def index
+    list_approvals
+  end
+
+  def edit
+  end
+
+  def show
+  end
+
+  def update
+    current_action = params[:approval_request][:current_action]
+    p "EEEEEE #{params[:approval_request][:current_action]}"
+    @approval_request.assign_attributes(
+      permitted_attributes(ApprovalRequest).merge(
+        status: current_action == "requested_for_changes" || current_action.blank? ? :published : :deleted,
+        last_action: current_action
+      )
+    )
+    @approval_request.save
+    @message =
+      Message.new(
+        message: params[:approval_request][:message],
+        author: current_user,
+        author_role: :mediator,
+        scope: :user_direct,
+        messageable: @approval_request
+      )
+    provider = @approval_request.approvable
+    case current_action
+    when "accepted"
+      Provider::Publish.call(provider)
```

Comment:
Assuming that publish, delete, unpublish actions on providers always succeeds, wrap them in some potential error handling with logging